app/assets/javascripts/client/controllers/responses/ResponseCtrl.js

```diff
@@ -1,14 +1,38 @@
-PDRClient.controller('ResponseCtrl', [
+(function() {
+  'use strict';
+  angular.module('PDRClient')
+      .controller('ResponseCtrl', ResponseCtrl);
+
+  ResponseCtrl.$inject = [
     '$scope',
     '$timeout',
-    'SessionService',
-    'Assessment',
     '$stateParams',
-    function($scope, $timeout, SessionService, Assessment, $stateParams) {
-      $scope.user = SessionService.getCurrentUser();
-
-      $scope.assessmentId = $stateParams.assessment_id;
-      $scope.responseId   = $stateParams.response_id;
-      $scope.assessment   = Assessment.get({id: $scope.assessmentId});
-    }
-]);
+    '$modal',
+    'Assessment',
+    'LearningQuestion'
+  ];
+
+  function ResponseCtrl($scope, $timeout, $stateParams, $modal, Assessment, LearningQuestion) {
+
+    $scope.assessmentId = $stateParams.assessment_id;
+    $scope.responseId = $stateParams.response_id;
+    $scope.assessment = Assessment.get({id: $stateParams.assessment_id});
+
+
+    $scope.$on('responses-loaded', function() {
+      $timeout(function() {
+        LearningQuestion.exists({assessment_id: $stateParams.assessment_id})
+            .$promise
+            .then(function() {
+              // No-op - implies that a learning question exists for this user on this assessment
+            }, function() {
```

Comment:
In the context of a HEAD request, we would either get back a 204 (No Content) or a 404 (Not Found), which are the only two states we should generally care about with this request.  That is to say, when we make the request, all we're checking to see is if we get a response back in the affirmative for the condition being true.

If we get back a 5xx error (most of the errors I've seen from Rails recently have been 500s, which is where my mind was at the time), then something on the server side has erred.  We shouldn't display the modal in the event of this, since we can't determine if the entity does or does not exist due to the error.